docs/best_practices/extensions.rst

```diff
@@ -45,3 +134,68 @@ anybody who receives the data also receives the necessary data to interpret it.
 .. note::
     In :pynwb-docs:`PyNWB <>`, the extension is cached automatically. This can be specified explicitly with
     ``io.write(filepath, cache_spec=True)``
+
+
+Use Attributes for small metadata related to a particular data object (Group or Dataset)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Attributes should be used mainly to store small metadata (usually less than 64 Kbytes) that
+is associated with a particular Group or Dataset. Typical uses of attributes are, e.g., to
```

Comment:
Just reading this, it's hard to grasp 'how big' something is to be less than 64 KB

Since most metadata are strings, maybe it would be useful to mention a relative scale to that?

A standard Python string under 64 KB would correspond to ~1300 characters, from 

```python
import sys

sys.getsizeof("a")
> 50

64_000 / 50
> 1280
```

Google also says the average number of characters per word in English is 4.7, so that's ~272 words per string

Reply:
Mostly relevant to fields like `description`, or sometimes `reference_frame`

Reply:
I like the idea of relating this to characters. Do you know how big characters are within HDF5? Is it the same as Python strings in memory?

Reply:
Hmm I'll have to investigate that a bit deeper, but most likely not the same as Python... I seem to remember us having some issues related to their byte string types here on the Inspector, but maybe that was for fields that are technically set as datasets, not attributes

Even in Python, if you apply a specific encoding it can change the size quite a bit

```python
sys.getsizeof("a".encode("utf-8"))
> 34
```
 which would put the max closer to ~400 words

Reply:
Well, from https://docs.h5py.org/en/stable/strings.html#storing-strings and https://docs.h5py.org/en/stable/special.html#h5py.string_dtype and manually inspecting some of these attributes with HDFView

It seems HDF5 stores with UTF-8 encoding, so about ~400 words on average from above

Reply:
> ~400 words

words or characters?

Reply:
> words or characters?

words using Google's estimate of average characters per word in English

Reply:
Or just shy of ~1900 characters if that's easier to communicate